eng/validation/build-arcadewithrepo.ps1

```diff
@@ -195,52 +210,105 @@ function Git-Command($repoName) {
+$darcGitHubRepoName = "https://github.com/${githubOrg}/${githubRepoName}"
+$darcAzDORepoName = "https://${azdoOrg}.visualstudio.com/${azdoProject}/_git/${azdoRepoName}"
 
-# check to see if branch exists and clean it up if it does
-$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha
+Write-Host "Last Known Good Build SHA: ${sha}"
+
+## Clone the repo from git
+Write-Host "Cloning '${githubRepoName} from GitHub"
+GitHub-Clone $githubRepoName
+ 
+## Check to see if branch exists and clean it up if it does
+$branchExists = $false
+if($true -eq $pushBranchToGithub)
+{
+    Write-Host "Looking up '${targetBranch}' branch on GitHub"
+    $branchExists = Git-Command $githubRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+}
+else 
+{
+    Write-Host "Looking up '${targetBranch}' branch on Azure DevOps"
+    $branchExists = Git-Command $githubRepoName ls-remote --heads $azdoUri refs/heads/$targetBranch
+}
 if($null -ne $branchExists)
 {
+    Write-Host "${targetBranch} was found. Attempting to clean up."
     try
     {
         & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
-        Git-Command $targetRepoName push origin --delete $targetBranch
+        if($true -eq $pushBranchToGithub)
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcGitHubRepoName --github-pat $githubPAT --password $barToken
+            Git-Command $githubRepoName push origin --delete $targetBranch
+        }
+        else
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcAzDORepoName --azdev-pat $azdoToken --password $barToken
+            Git-Command $githubRepoName remote add $remoteName $azdoUri
+            Git-Command $githubRepoName push $remoteName --delete $targetBranch
+        }
     }
     catch
     {
-        Write-Warning "Unable to delete default or branch when cleaning up existing branch"
+        Write-Warning "Unable to delete default channel or branch when cleaning up existing branch"
     }
 }
-Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Create a branch from the repo with the given SHA.
+Git-Command $githubRepoName checkout -b $targetBranch $sha
 
 ## Make the changes to that branch to update Arcade - use darc
-Set-Location $(Get-Repo-Location $targetRepoName)
-& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+Set-Location $(Get-Repo-Location $githubRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --azdev-pat $azdoToken --password $barToken
```

Comment:
I think this has the possibility of updating to a newer build than is currently running in arcade-validation, if the validation channel gets updated while this build is running. If we have the BAR id of the input build, you could use ```update-dependencies --id```

Reply:
Oh yeah, you're right. And this is actually a pretty big chance. Would give us results for the wrong arcade build.

Reply:
If you have the arcade version you could look up the BAR build id with get-asset

Reply:
Did this. Let me know if the darc call has superfluous parameters or if it's okay the way it is

Reply:
You can drop the repo and channel now that you specify the build